### ISSUE TITLE
Towards Open Set Deep Networks

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#358]
https://www.cv-foundation.org/openaccess/content_cvpr_2016/papers/Bendale_Towards_Open_Set_CVPR_2016_paper.pdf

**Comment by sweaterr** [comment]
# 1 Introduction

**Comment by sweaterr** [comment]
Our approach is based on characterizing known classes with extreme value signatures (EVS) computed for neural activation vectors.
우리의 접근법은 신경 활성화 벡터에 대해 계산 된 극단 값 서명 (EVS)으로 알려진 클래스의 특성을 기반으로합니다.

Extreme value signatures specify which dimensions of deep neural activations have largest values.
극단 값 서명은 가장 큰 값을 갖는 심 신경 활성화의 차원을 지정합니다.

Previous work has shown that a surprisingly large fraction of the original image signal can be recovered from the high-scored activations [16].
이전의 연구 결과는 원래 이미지 신호의 놀랍게도 많은 부분이 높은 점수를받은 활성화 [16]로부터 복구 될 수 있음을 보여주었습니다.

We go one step further and show that the signatures even capture discriminative information not only with respect to known but also with respect to unknown classes.
우리는 한걸음 더 나아가 알려지면서 알려진 클래스뿐만 아니라 알려지지 않은 클래스와 관련하여 차별 정보를 포착하기까지합니다.

EVS are therefore suited for novelty detection as visualized in Fig  1.
따라서 EVS는 그림 1에서 시각화 된 참신 탐지에 적합합니다.

For an efficient realization, we follow a prototype-based approach, where each known class is encoded by the EVS of the class-mean activation vector.
효율적인 구현을 위해 프로토 타입 기반 접근 방식을 따르며, 각 알려진 클래스는 클래스 평균 활성화 벡터의 EVS로 인코딩됩니다.

In summary, we found that our extreme value signatures are compact, effective for novelty detection, and a promising representation for further improvements of the terra incognita of open-set recognition.
요약하면, 우리는 극단적 인 가치의 서명이 신기원 탐지를 위해 작고 효과적이며 오픈 세트 인식에 대한 테라 인크루먼트 (terra incognita)의 추가 개선에 대한 유망한 대표성을 발견했습니다.

**Comment by sweaterr** [comment]
## 3 Extreme Value Signatures

Neural Activations for Novelty Detection Many popular approaches in machine vision use activations of pre-trained CNNs as off-the-shelf image representations [7,23].
참신 탐지를위한 신경 활성화 머신 비전에서 널리 사용되는 접근법은 기성의 이미지 표현으로 사전 훈련 된 CNN의 활성화를 사용합니다 [7,23].

Our current understanding is that the majority of trainable layers in CNNs (especially convolutional layers) becomes sensitive to the specific characteristics that are common in natural images when fed with millions of training images.
우리의 현재의 이해는 CNN (특히 길쌈 레이어)의 훈련 가능한 레이어의 대부분은 수백만 개의 교육 이미지가 제공 될 때 자연스러운 이미지에서 일반적인 특성에 민감하게 반응한다는 것입니다.

In particular, it has been shown that filter masks of convolutional layers become sensitive to visual “elements” that can be commonly found in natural images [28].
특히, 콘볼 루션 레이어의 필터 마스크는 자연스러운 이미지에서 흔히 볼 수있는 시각적 "요소"에 민감하게 반응하는 것으로 나타났습니다 [28].

Whereas these elements mainly correspond to low-level texture patterns for lower layers, they can be related to semantically meaningful object parts in higher layers [24].
이 요소들은 주로 하위 계층의 낮은 수준의 텍스처 패턴에 해당하는 반면 상위 계층의 의미 적으로 의미있는 객체 부분과 관련 될 수 있습니다 [24].

This data-driven approach to representation learning is obviously appealing in several aspects, especially when large amounts of data are available [20].
표현 학습에 대한 이러한 데이터 중심 접근 방식은 특히 많은 양의 데이터를 사용할 수있는 경우 여러 측면에서 분명히 매력적입니다 [20].

Although the respective CNNs have originally been trained for discriminating among known categories, we show that their extreme value statistics can also be used to detect instances of novel categories.
각각의 CNN은 원래 알려진 카테고리를 구분하기 위해 원래 교육 받았지만 극한값 통계를 사용하여 새로운 카테고리의 인스턴스를 감지 할 수 있음을 보여줍니다.

**Comment by sweaterr** [comment]
### Extreme Value Signatures for Individual Images

Let x ∈ R D be the activation extracted at a chosen layer when applying a given CNN to a single image.
하나의 이미지에 주어진 CNN을 적용 할 때 선택된 레이어에서 추출 된 활성화를 x∈ R D 라하자.

Babenko et al, showed in [1] that this representation still maintains a large amount of discriminative information after binarization with a fixed threshold τ.
Babenko 등 [1]은 고정 된 임계 값 τ로 이진화 한 후에도 여전히 많은 양의 식별 정보를 유지하고 있음을 보였다.

Hence, the binary vector b:.
따라서, 바이너리 벡터 b :.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39093840-fa218730-4660-11e8-9daa-e1f6578becd0.png)

**Comment by sweaterr** [comment]
can serve as a substitute for x when the threshold τ is chosen appropriately since it translates x into a binary representation.
은 임계 값 τ가 x를 이진 표현으로 변환하기 때문에 적절하게 선택 될 때 x에 대한 대용 물로서 작용할 수있다.

Note that for the ease of readability, x [d] denotes the d th dimension of x and the function δ (v) maps to 1 or 0 if v is true or false, respectively.
가독성을 위해 x [d]는 x의 d 차원을 나타내고 함수 δ (v)는 v가 참 또는 거짓이면 각각 1 또는 0으로 매핑됩니다.

Despite the successful application of binarized neural activations in [1], relying on a constant threshold might be too restrictive for the complexity of visual recognition tasks.
[1]의 2 치화 된 신경 활성화의 성공적인 적용에도 불구하고, 일정한 임계 값에 의존하는 것은 시각적 인식 작업의 복잡성에 너무 제한적일 수 있습니다.

In a more general setting, we can replace this constant threshold by a threshold function T (·) which returns a threshold specifically tailored to each example x:.
좀 더 일반적인 설정에서이 상수 임계 값을 각 예제 x :에 맞게 특별히 고안된 임계 값을 반환하는 임계 함수 T (·)로 바꿀 수 있습니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39093845-0bf11624-4661-11e8-9964-15ff2afe9f49.png)

**Comment by sweaterr** [comment]
Our extreme value signature follows this general concept.
우리의 극단 값 서명은이 일반적인 개념을 따릅니다.

Let us therefore denote πx the permutation that brings x into descending order:.
그러므로 우리는 x를 내림차순으로 가져 오는 순열을 πx라고하자.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39093850-1b66473c-4661-11e8-8b2f-7c34d4cef59d.png)

**Comment by sweaterr** [comment]
Thereby, we obtain a threshold Trank K (x) for each x using the K-highest activation:

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39093852-24501ff8-4661-11e8-8da4-029897b70b68.png)

**Comment by sweaterr** [comment]
The resulting binary vector b (x, Trank K(x)) can therefore be seen as an indicator for the K highest values of x, which we refer to as the EVS of x.
결과로 생성 된 이진 벡터 b (x, Trank K (x))는 x의 EV 값을 나타내는 K의 최대 값을 나타내는 지표로 볼 수 있습니다.

When thinking of visual recognition scenarios, we expect that two images that contain similar visual concepts also lead to a similar set of extreme dimensions in the resulting neural activations Hence, their binarized codes based on Trank K should be close as well.
시각적 인식 시나리오를 생각할 때 유사한 시각적 개념을 포함하는 두 개의 이미지가 결과적으로 신경 활성화에 비슷한 극단적 인 차원을 초래할 것으로 예상합니다. 따라서 Trank K를 기반으로하는 이진 코드도 비슷해야합니다.

Therefore, we can apply the inner product.
따라서 내부 제품을 적용 할 수 있습니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39093859-3a926276-4661-11e8-98de-ca582957a4d6.png)

**Comment by sweaterr** [comment]
## 4 Experiments

We investigated our approach for multi-class novelty detection both quantitatively in comparison with state-of-the-art techniques (Sections 4.2 to 4.4) and qualitatively (Section 4.5).
우리는 최첨단 기술 (섹션 4.2 ~ 4.4) 및 정량적 (섹션 4.5)과 비교하여 정량적으로 다중 클래스 신규성 탐지에 대한 접근 방식을 조사했습니다.

As benchmarking set, we chose the popular ImageNet dataset from ILSVRC’12 [20] and derived novelty detection tasks of varying difficulty.
벤치마킹 세트로서 ILSVRC'12 [20]에서 인기있는 ImageNet 데이터 세트를 선택하고 다양한 난이도의 참신 탐지 작업을 유도했습니다.

In all experiments, we encode images with neural activations of a AlexNet-Places365-CNN [29] without any fine-tuning.
모든 실험에서 우리는 미세 조정없이 AlexNet-Places 365-CNN [29]의 신경 활성화를 사용하여 이미지를 인코딩합니다.

Thereby, none of the involved classes was already observed during model training1.
따라서 모델 훈련 중에 이미 관련된 수업은 관찰되지 않았습니다 1.

We experimented with activations from layers CONV4 to FC8 with and without RELU and feature normalization to unit length.
우리는 RELU의 유무에 관계없이 레이어 CONV4에서 FC8까지의 활성화와 유닛 길이에 대한 피쳐 정상화를 실험했습니다.

In the following, we only show results for FC6 which lead to highest overall accuracy, but provide evaluations for all layers in the supplementary material (S.1).
다음은 전체 정확도가 가장 높은 FC6의 결과 만 보여 주지만 보충 자료 (S.1)의 모든 레이어에 대한 평가를 제공합니다.
